tests/webTables.spec.ts

```diff
@@ -0,0 +1,146 @@
+import test, { expect } from "@playwright/test"
+
+
+test.describe('Web Tables', () => {
+    test.beforeEach(async ({ page }) => {
+        await page.goto("/")
+    })
+
+    test('Test Case 1: Validate the pet name city of the owner', async ({ page }) => {
+        await page.getByRole('button', { name: 'Owners' }).click()
+        await page.getByRole('link', { name: 'Search' }).click()
+        await page.getByRole('link', { name: 'Jeff Black' }).click()
+        await expect(page.locator('.ownerFullName')).toHaveText('Jeff Black')
+        const petLuckySection = page.locator('.dl-horizontal', { hasText: "Lucky" })
+        await expect(petLuckySection).toContainText('Lucky')
+        await expect(page.locator('td', { hasText: 'Monona' })).toHaveText('Monona')
+    })
+
+    test('Test Case 2: Validate owners count of the Madison city', async ({ page }) => {
+        await page.getByRole('button', { name: 'Owners' }).click()
+        await page.getByRole('link', { name: 'Search' }).click()
+        await expect(page.getByRole('row', { name: 'Madison' })).toHaveCount(4)
+    })
+
+
+
+    test('Test Case 3: Validate search by Last Name', async ({ page }) => {
+        await page.getByRole('button', { name: 'Owners' }).click()
+        await page.getByRole('link', { name: 'Search' }).click()
+        await page.locator('#lastName').fill('Black')
+        await page.getByRole('button', { name: 'Find Owner' }).click()
+        await expect(page.locator('tbody').getByRole('cell', { name: 'Jeff Black' })).toContainText('Black')
+        await page.locator('#lastName').clear()
+        await page.locator('#lastName').fill('Davis')
+        await expect(page.locator('tbody').getByRole('cell', { name: 'Betty Davis' })).toContainText('Davis')
+        const lastNames = ["Es", "Playwright"]
+
+        for (const lastName of lastNames) {
+            await page.getByRole('textbox').fill(lastName)
```

Comment:
Be consistent with locator selection. Earlier you have used ID for this field, now it's getByRole()